### PR TITLE
Output distances in kdtree

### PR DESCRIPTION
**Pull request by msluszniak**

### CODE REVIEW

**Review by josevalim — APPROVED** [review]

File: lib/scholar/neighbors/kd_tree.ex [hunk 1]
```diff
@@ -266,26 +306,28 @@ defmodule Scholar.Neighbors.KDTree do
...
+        [
+          [2.0, 2.0, 6.0],
+          [7.0, 7.0, 7.0],
+          [3.0, 3.0, 5.0]
+        ]
+      >
   """
-  deftransform predict(tree, data, opts \\ []) do
-    predict_n(tree, data, NimbleOptions.validate!(opts, @predict_schema))
+  deftransform predict(tree, data) do
```
Comment (krstopro) [inline lib/scholar/neighbors/kd_tree.ex:319]: `deftransform` is redundant here, isn't it? I would add data validation inside this function though.

Comment (krstopro) [reply lib/scholar/neighbors/kd_tree.ex:319]: Oh, my bad, I now see that the options are validated. I guess the options to `predict` should be removed.

Comment (msluszniak) [reply lib/scholar/neighbors/kd_tree.ex:319]: Yeah, you're right. As we now validate options in `fit` we can make it `defn`

Comment (krstopro) [reply lib/scholar/neighbors/kd_tree.ex:319]: I would validate `data` in predict as well, maybe rename it to `query` (up to you).
Just the usual stuff: that rank is equal to 2 and that the 2nd dimension matches the one of the data used to fit the model.

**Pull request merged**